src/kernel/tty.zig

```diff
@@ -59,15 +58,19 @@ pub fn print(comptime format: []const u8, args: var) void {
 /// Clear the screen by printing a space at each cursor position. Sets the cursor to the top left (0, 0)
 ///
 pub fn clear() void {
-    var row: u8 = 0;
-    while (row < tty.rows) : (row += 1) {
-        var col: u8 = 0;
-        while (col < tty.cols) : (col += 1) {
-            print(" ", .{});
+    if (tty.clear) |clr| {
+        clr();
+    } else {
+        var row: u8 = 0;
+        while (row < tty.rows) : (row += 1) {
+            var col: u8 = 0;
+            while (col < tty.cols) : (col += 1) {
+                print(" ", .{});
+            }
+            print("\n", .{});
         }
-        print("\n", .{});
+        tty.setCursor(0, 0);
```

Comment:
An idea. As there will be many calls to print, this will be slow, so was wondering if you could make some of it into comptime, like calculate the number of spaces and then print those spaces for the number or rows. like:
```Zig
const spaces = " " * tty.cols;
print("{}", .{spaces});
```
Could also do the same for the new line, so make the string then call print once

Reply:
To avoid making rows and cols comptime (I don't think that would be possible considering how initTTY et.c works), I could allocate a row of spaces, print that n times then free the row.